cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/UninstallUtils.java

```diff
@@ -0,0 +1,120 @@
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.packerina;
+
+import org.ballerinalang.compiler.BLangCompilerException;
+import org.wso2.ballerinalang.compiler.util.ProjectDirConstants;
+import org.wso2.ballerinalang.util.RepoUtils;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * This class provides util methods for uninstalling Ballerina packages.
+ *
+ * @since 0.982.0
+ */
+public class UninstallUtils {
+    private static final Path BALLERINA_HOME_PATH = RepoUtils.createAndGetHomeReposPath();
+    private static PrintStream outStream = System.out;
+
+    /**
+     * Uninstalls the package from the home repository.
+     *
+     * @param fullPkgPath package provided
+     */
+    public static void uninstallPackage(String fullPkgPath) {
+        String orgName;
+        String packageName;
+        String version;
+        // Get org-name
+        int orgNameIndex = fullPkgPath.indexOf("/");
+        if (orgNameIndex == -1) {
+            throw new BLangCompilerException("no org-name is provided");
+        }
+        orgName = fullPkgPath.substring(0, orgNameIndex);
+
+        // Get package name
+        int packageNameIndex = fullPkgPath.indexOf(":");
+        if (packageNameIndex == -1) { // version is provided
+            throw new BLangCompilerException("no package version is provided");
+        }
+        packageName = fullPkgPath.substring(orgNameIndex + 1, packageNameIndex);
+        version = fullPkgPath.substring(packageNameIndex + 1, fullPkgPath.length());
+
+        Path homeRepoPath = BALLERINA_HOME_PATH.resolve(ProjectDirConstants.DOT_BALLERINA_REPO_DIR_NAME);
+        Path cacheRepoPath = BALLERINA_HOME_PATH.resolve(ProjectDirConstants.CACHES_DIR_NAME)
+                                                .resolve(ProjectDirConstants.BALLERINA_CENTRAL_DIR_NAME);
+        Path pkgPath = Paths.get(orgName, packageName, version);
+
+        // Check if package is installed locally
+        if (Files.exists(homeRepoPath.resolve(pkgPath), LinkOption.NOFOLLOW_LINKS)) {
+            deletePackage(homeRepoPath, pkgPath, fullPkgPath, packageName);
+        } else if (Files.exists(cacheRepoPath.resolve(pkgPath), LinkOption.NOFOLLOW_LINKS)) {
+            deletePackage(cacheRepoPath, pkgPath, fullPkgPath, packageName);
+        } else {
+            // package to be uninstalled doesn't exists
+            throw new BLangCompilerException("package does not exist " + fullPkgPath);
+        }
+    }
+
+    /**
+     * Remove package from home repository.
+     *
+     * @param repoPath    path of the home repository
+     * @param pkgPath     package path
+     * @param fullPkgPath full package path as given by user
```

Comment:
This is more of a package ID isn't it ? But have we used package path across other implementations ?

Reply:
Also mention that it is for logging purposes only.

Reply:
This is not the Package ID. fullPkgPath is the package to be uninstalled which is provided by the user. I will add a comment that it is used only for logging purposes.